### PR TITLE
OPS Backend - Step 2

### PR DESCRIPTION
**Pull request by btrb**
This PR is for populating OPS calls over the IET through the method `_specialize_iet` within the the class `Operator`

### CODE REVIEW

**Comment by btrb** [comment]
Maybe some rebasing would be needed for starting a new commit tree. Sorry for the mess.

File: devito/ops/operator.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+__all__ = ['Operator']
+
+
+class Operator(OperatorRunnable):
+    """
+    A special :class:`OperatorCore` to JIT-compile and run operators through OPS.
+    """
+
+    def _specialize_iet(self, iet, **kwargs):
+        # print(iet)
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: drop spurious, commented-out prints

File: devito/ops/operator.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        grid_decl = build_grid_decl(iet)
+        # print(grid_decl)
+
+        iet = List(body=grid_decl + [iet])
+        print(iet)
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: still raise NotImplementedError at the end?

File: devito/ops/operator.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+
+def build_grid_decl(iet):
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: we need specific tests for this function, to be put in ``test_ops.py``

File: devito/yask/operator.py [hunk 4]
```diff
@@ -100,6 +99,7 @@ def _specialize_iet(self, iet, **kwargs):
                      yc_soln.get_num_equations()))
             except NotImplementedError as e:
                 log("Unable to offload a candidate tree. Reason: [%s]" % str(e))
+
```
Comment (FabioLuporini) [inline devito/yask/operator.py]: ?

File: devito/yask/operator.py [hunk 5]
```diff
@@ -110,6 +110,7 @@ def _specialize_iet(self, iet, **kwargs):
         # in storage layout employed by Devito and YASK
         yk_grid_objs = {i.name: YaskGridObject(i.name) for i in self.input if i.from_YASK}
         yk_grid_objs.update({i: YaskGridObject(i) for i in self._local_grids})
+
```
Comment (FabioLuporini) [inline devito/yask/operator.py]: ?

**Comment by btrb** [comment]
[self-reminder] as said in the last commit msg: Most of the OPS calls and declarations have been built through Devito objects. Now we shift for a new branch to implement an "OPS Kernel Builder" from SymPy Exps.

**Pull request closed without merging**